core/src/main/java/org/elasticsearch/gateway/TransportNodesListGatewayStartedShards.java

```diff
@@ -139,16 +143,17 @@ protected NodeGatewayStartedShards nodeOperation(NodeRequest request) {
         }
     }
 
-    private boolean canOpenIndex(ShardId shardId, IndexMetaData metaData) throws IOException {
-        // try and see if we an list unallocated
-        if (metaData == null) {
-            return false;
-        }
+    private void tryOpenIndex(ShardId shardId, IndexMetaData metaData) throws IOException {
         final ShardPath shardPath = ShardPath.loadShardPath(logger, nodeEnv, shardId, metaData.settings());
         if (shardPath == null) {
-            return false;
+            throw new IllegalStateException(shardId + " no shard state found");
+        }
+        try {
+            Store.tryOpenIndex(shardPath.resolveIndex());
+        } catch (Exception exception) {
+            logger.trace("Can't open index for path [{}]", exception, shardPath.resolveIndex());
```

Comment:
I think this results in double logging, we will log this here too, no? https://github.com/elastic/elasticsearch/pull/11545/files#diff-9a7b643d56430763f42de913657ca798R125